name.abuchen.portfolio/src/name/abuchen/portfolio/util/HolidayType.java

```diff
@@ -30,6 +30,353 @@ public LocalDate apply(LocalDate date)
         }
     }
 
+    private static class JewishCalendarDate
+    {
+        JewishCalendarDate(int day, int month, int year)
+        {
+            this.day = day;
+            this.month = month;
+            this.year = year;
+        }
+
+        // JewishCalendarDate(JewishCalendarDate date)
+        // {
+        // this.day = date.getDay();
+        // this.month = date.getMonth();
+        // this.year = date.getYear();
+        // }
+
+        public int getDay()
+        {
+            return day;
+        }
+
+        public int getMonth()
+        {
+            return month;
+        }
+
+        public int getYear()
+        {
+            return year;
+        }
+
+        @SuppressWarnings("unused")
+        public void setDay(int day)
+        {
+            this.day = day;
+        }
+
+        @SuppressWarnings("unused")
+        public void setMonth(int month)
+        {
+            this.month = month;
+        }
+
+        @SuppressWarnings("unused")
+        public void setYear(int year)
+        {
+            this.year = year;
+        }
+
+        @SuppressWarnings("unused")
+        public boolean areDatesEqual(JewishCalendarDate date)
+        {
+            if ((day == date.getDay()) && (month == date.getMonth()) && (year == date.getYear()))
+                return true;
+            return false;
+        }
+
+        @SuppressWarnings("unused")
+        public int getHashCode()
+        {
+            return (year - 1583) * 366 + month * 31 + day;
+        }
+
+        @Override
+        public String toString()
+        {
+            return day + "." + month + "." + year; //$NON-NLS-1$//$NON-NLS-2$
+        }
+
+        private int day;
+        private int month;
+        private int year;
+    }
+
+    private static class CalendarImpl
```

Comment:
Maybe it makes sense to extract this class a package private class JewishCalendar in the same package. (Back when creating this class, I was a fan of private classes to keep the implementations hidden, but as the file is growing, then it could make sense to keep it separate).